src/core/service.c

```diff
@@ -1532,6 +1532,35 @@ static int service_spawn_internal(
                 log_unit_debug(UNIT(s), "Passing %zu fds to service", exec_params.n_socket_fds + exec_params.n_storage_fds);
         }
 
+        /* These options require PrivateUsers= when used in user units, as we need to be in a user namespace
+         * to have permission to enable them when not running as root. */
+        if (!s->exec_context.private_users && !MANAGER_IS_SYSTEM(UNIT(s)->manager))
```

Comment:
Is this only for service units? Why?

Reply:
It's for `[Service]` sandboxing options? Where else would it be used?